### ISSUE TITLE
utils.read_molecules method deletes molecule title?

### ISSUE THREAD

**Issue by bannanc** [open-forcefield-group/smarty#209]
While trouble shooting smarty with MiniDrugBank, I was trying to see which molecule was failing to be typed. 

I noticed something. If I read in a molecules file (MiniDrugBank_atyped.oeb) myself, such as the script below, it keeps titles I assigned the molecules. But if I used the utils.read_molecules method it loses the titles. 

Self loading:
```
load_mols = list()
inf = "MiniDrugBank_atyped.oeb"
ifs = oemolistream(inf)
mol = OEMol()

while OEReadMolecule(ifs,mol):
    load_mols.append(OEMol(mol))
ifs.close()
print(load_mols[0].GetTitle())
# prints: "MiniDrugBank_000"
```

Read_Molecules:
```
load_mols = read_molecules("MiniDrugBank_atyped.sdf")
print("'%s'" % load_mols[0].GetTitle())
# prints: ''
```

Inside read_molecules there is this line:
```
# Get molecule name.
name = OEGetSDData(molecule, 'name').strip()
molecule.SetTitle(name)
```
This seems to erase the molecule title unless there is specifically an SD tag with the name. 

Even if I set titles and write out molecules to an SDF file the read_molecules method still removes titles. 
I would like to remove the SetTitle lines or only use them if GetTitle() is empty?
@davidlmobley fyi

**Comment by davidlmobley** [comment]
Yes, I don't see any reason we should be overwriting existing titles.

**Comment by bannanc** [comment]
Ok, I'll include it with the still open pull request.

**Comment by bannanc** [comment]
Fixed with pull request #198
